destination_sdk.proto

```diff
@@ -55,7 +55,26 @@ message CreateTableResponse {
 message AlterTableRequest {
     map<string, string> configuration = 1;
     string schema_name = 2;
-    Table table = 3;
+    string table_name = 3;
+    repeated SchemaDiff changes = 4;
```

Comment:
Do our own DataWriter implementations get something like this? How is it generated?

Reply:
Yes, we do follow similar approach. The [updateSchema](https://github.com/fivetran/engineering/blob/main/core_interfaces/src/com/fivetran/core/interfaces/writer/DataWriter.java#L79) method in DataWriter generates the schema differences and updates the schema.
For eg: In SQLLikeDataWriter, we compute the difference in primary key, update in column type, added new columns etc - [SQLLikeDataWriter#updateSchema](https://github.com/fivetran/engineering/blob/c84744d811da77d79dbb609a97c9f7291bb33bad/warehouses/common/src/com/fivetran/warehouses/common/SqlLikeDataWriter.java#L493-L543).

Reply:
@georgewfraser no this is something new we introduced. `SdkDataWriter` generates the diffs to pass to the partner code so they don't have to do it on their own. It would be great if we can move this to the core so the other data writers didn't have to perform the diff either. (In fact, the core doesn't even do a simple compare between existing and new schema, calling `DataWriter#updateSchema` whether the two are different or not, expecting every data writer to figure it out)